tests/Doctrine/ODM/MongoDB/Tests/Functional/EnumTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ODM\MongoDB\Tests\Functional;
+
+use Doctrine\ODM\MongoDB\Mapping\Driver\AttributeDriver;
+use Doctrine\ODM\MongoDB\Tests\BaseTest;
+use Doctrine\Persistence\Mapping\Driver\MappingDriver;
+use Documents81\Card;
+use Documents81\Suit;
+use MongoDB\BSON\ObjectId;
+use ValueError;
+
+use function assert;
+use function sprintf;
+
+/**
+ * @requires PHP 8.1
+ */
+class EnumTest extends BaseTest
+{
+    public function testPersistNew(): void
+    {
+        $doc       = new Card();
+        $doc->suit = Suit::Clubs;
+
+        $this->dm->persist($doc);
+        $this->dm->flush();
+        $this->dm->clear();
+
+        $saved = $this->dm->find(Card::class, $doc->id);
+        assert($saved instanceof Card);
```

Comment:
```suggestion
        $this->assertInstanceOf(Card::class, $saved);
```
To be consistent with other tests